scripts/install_dependencies/install_NEO_OCL_driver.sh

```diff
@@ -524,24 +609,32 @@ check_agreement()
 check_specific_generation()
 {
     echo "Checking processor generation..."
-    specific_generation=$(grep -m1 'model name' /proc/cpuinfo | grep -E "i[357]-1[01][0-9]{2,4}N?G[147R]E?")
-    if [[ ! -z "$specific_generation" && "$INSTALL_DRIVER_VERSION" != '20.35.17767' ]]; then
-        echo "$(basename "$0"): Detected 10th generation Intel® Core™ processor (formerly Ice Lake) or 11th generation Intel® Core™ processor (formerly Tiger Lake)."
-        echo "Driver version 20.35.17767 is going to be installed to fully utilize hardware features and performance."
-        if [ "$auto_mode" == true ]; then
-            INSTALL_DRIVER_VERSION='20.35.17767'
-            return 0
-        else
-            while true; do
-                read -p "You are still able to use the older version 19.41.14441. Use the older driver? (y/n) [n] " yn
-                yn=${yn:=n}
-                case $yn in
-                    [Yy]*) return 0 ;;
-                    [Nn]*) INSTALL_DRIVER_VERSION='20.35.17767' && return 0 ;;
-                esac
-            done
+    typeset -A specific_generation
```

Comment:
Perhaps this function can be simplified in order to avoid arrays. If it would accept version and regexp as parameters, then whole code could be slightly more readable:

Function calls (2 instead of 1):
```.sh
check_specific_generation "20.35.17767" "i[357]-1[01][0-9]{2,4}N?G[147R]E?" "10th generation Intel® Core™ processor (formerly Ice Lake) or 11th generation Intel® Core™ processor (formerly Tiger Lake)."
check_specific_generation "21.29.20389" "i[3579]-12[0-9]{3}[A-Z]{0,2}" "12th generation Intel® Core™ processor (formerly Alder Lake)."
```
Function body (arrays and for loop are not needed):
```
check_specific_generation()
{
    driver = $1
    regexp = $2
    desc = $3
    # ...
}
```

Reply:
It's corner case for 2021.4.1 release.
I think we can keep it as is.